src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/resource/ActivitiesResource.kt

```diff
@@ -68,4 +76,56 @@ class ActivitiesResource(private val activitiesService: ActivitiesService) {
     @RequestBody @Valid createActivityRequest: CreateActivityRequest
   ): CreateActivityResponse =
     activitiesService.createActivity(createActivityRequest)
+
+  @PreAuthorize("hasRole('ROLE_NOMIS_ACTIVITIES')")
+  @PostMapping("/activities/offender-program-profile")
+  @ResponseStatus(HttpStatus.CREATED)
+  @Operation(
+    summary = "Allocates a prisoner to an activity",
+    description = "Allocates a prisoner to an activity. Requires role NOMIS_ACTIVITIES",
+    requestBody = io.swagger.v3.oas.annotations.parameters.RequestBody(
+      content = [
+        Content(
+          mediaType = "application/json",
+          schema = Schema(implementation = CreateOffenderProgramProfileRequest::class)
+        )
+      ]
+    ),
+    responses = [
+      ApiResponse(
+        responseCode = "201",
+        description = "Offender program profile information with created id",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = CreateOffenderProgramProfileResponse::class)
+          )
+        ]
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Activity or booking does not exist",
```

Comment:
Should that be on a 404?

Reply:
Dont think so , its bad data because the related course activity id or offender booking is invalid

Reply:
Ah never mind, I've seen 404s used elsewhere (in PrisonAPI?) but thinking about a 400 is probably right.

Reply:
I think if we add courseActivityId to URL then a 404 but 400 for missing booking since it is in the body

Reply:
done